core/ir/arm/decode.c

```diff
@@ -172,7 +174,8 @@ decode_in_it_block(decode_state_t *state, app_pc pc, decode_info_t *di)
 bool
 is_isa_mode_legal(dr_isa_mode_t mode)
 {
-    return (mode == DR_ISA_ARM_THUMB || DR_ISA_ARM_A32);
```

Comment:
This would always return `true`.
I don't think this was the intended behavior, so I fixed it and added `DR_ISA_SYNTHETIC` to the "legal modes" of arm.

Reply:
Agreed, looks like it was a bug.